source/geod24/LocalRest.d

```diff
@@ -1594,3 +1589,46 @@ unittest
         assert(ex.msg == `"Request timed-out"`);
     }
 }
+
+unittest
+{
+    static import geod24.concurrency;
+
+    __gshared C.Tid node_tid;
+
+    static interface API
+    {
+        void segfault ();
+        void check ();
+    }
+
+    import std.stdio;
+
+    static class Node : API
+    {
+        override void segfault ()
+        {
+            int* ptr; *ptr = 1;
+        }
+
+        override void check ()
+        {
+            auto node = new RemoteAPI!API(node_tid);
+
+            // We need to return immediately so that the main thread can continue testing
+            runTask(() {
+                node.ctrl.sleep(500.msecs);
+                node.segfault();
+            });
+        }
+    }
+
+    auto node_1 = RemoteAPI!API.spawn!Node();
+    auto node_2 = RemoteAPI!API.spawn!Node();
+    node_tid = node_2.tid;
+    node_1.check();
+    node_2.ctrl.shutdown();  // shut it down before wake-up, segfault() command will be ignored
+
+    Thread.sleep(1.seconds);
```

Comment:
What's the intent of the sleep ? To make sure the Thread terminates ? If so joining the thread might be better no ?